core/src/main/java/org/opensearch/sql/data/model/ExprDateValue.java

```diff
@@ -36,9 +43,44 @@ public ExprDateValue(String date) {
     }
   }
 
+  /** Constructor of ExprDateValue to support custom/OpenSearch date formats in mappings. */
+  public ExprDateValue(String date, List<DateFormatter> dateFormatters) {
+    LocalDate localDate = null;
+    String dateFormat = "";
+    // check if dateFormatters are empty, then set default ones
+    if (dateFormatters == null || dateFormatters.isEmpty()) {
+      List<String> defaultPatterns =
+          Arrays.asList("strict_date_time_no_millis", "strict_date_optional_time", "epoch_millis");
+      dateFormatters = new ArrayList<>();
+      for (String pattern : defaultPatterns) {
+        dateFormatters.add(DateFormatter.forPattern(pattern));
+      }
+    }
+    for (DateFormatter formatter : dateFormatters) {
+      try {
+        TemporalAccessor accessor = formatter.parse(date);
+        ZonedDateTime zonedDateTime = DateFormatters.from(accessor);
+        localDate = zonedDateTime.withZoneSameLocal(ZoneOffset.UTC).toLocalDate();
+
+        dateFormat = formatter.format(accessor);
+        break;
+      } catch (IllegalArgumentException ignored) {
+        // nothing to do, try another format
+      }
```

Comment:
What happens if all the formatter failed? Should we throw exception in that case?

Reply:
Yes if we cannot parse using default/custom or constructor set formats, we are throwing exception on line #70.